### PR TITLE
tasks: list tasks with proper alignment

### PR DESCRIPTION
**Pull request by kraptor**
This is a small QoL patch.

Currently, `nimble` lists tasks without proper alignment between the name and the description of the task:

```
$ nimble tasks
build_debug        Build debug version
build_release        Build release version
build_profiler        Build with profiler
build_all        Build all versions
clean        Clean all build files
```

This patch adds the ability to collect the max length of the tasks names (with a minimum of 8 characters), then render descriptions properly aligned:

```
$ ./nimble tasks
build_debug     Build debug version
build_release   Build release version
build_profiler  Build with profiler
build_all       Build all versions
clean           Clean all build files
```

I tried to keep the same code style as the code aroudn. Please, let me know if there is anything else you'd like to improve/change.

**NOTE**: this change may *potentially* break scripts dealing with `nimble tasks` output, as it does inject two spaces between task name and task description, instead of a bunch of spaces like before.

The following tests did not pass on master before, not sure if because of my installation of Nim (using 1.5.1-latest) or my system:
* nimscript evaluation error message

### CODE REVIEW

File: src/nimblepkg/nimscriptapi.nim [hunk 1]
```diff
@@ -178,10 +184,12 @@ template task*(name: untyped; description: string; body: untyped): untyped =
   proc `name Task`*() = body
 
   nimbleTasks.add astToStr(name)
+  if astToStr(name).len > nimbleTasksMaxNameLen:
+    nimbleTasksMaxNameLen = astToStr(name).len
```
Comment (dom96) [inline src/nimblepkg/nimscriptapi.nim:188]: Nit: you can do this in `onExit` and save the need for another global variable

File: src/nimblepkg/nimscriptapi.nim [hunk 2]
```diff
@@ -178,10 +184,12 @@ template task*(name: untyped; description: string; body: untyped): untyped =
...
+  nimbleTasksDescriptions.add description
```
Comment (dom96) [inline src/nimblepkg/nimscriptapi.nim:189]: Maybe also turn `nimbleTasks` into a list of tuples to keep the names and descriptions together?

File: src/nimblepkg/nimscriptapi.nim [hunk 3]
```diff
@@ -216,3 +224,6 @@ proc getPkgDir*(): string =
   result = projectFile.rsplit(seps={'/', '\\', ':'}, maxsplit=1)[0]
 
 proc thisDir*(): string = getPkgDir()
+
+
+
```
Comment (dom96) [inline src/nimblepkg/nimscriptapi.nim:229]:
```suggestion
```

**Review by dom96 — CHANGES_REQUESTED** [review]

File: src/nimblepkg/nimscriptapi.nim (hunk 3 above)
Comment (kraptor) [reply src/nimblepkg/nimscriptapi.nim]: Done.

File: src/nimblepkg/nimscriptapi.nim (hunk 2 above)
Comment (kraptor) [reply src/nimblepkg/nimscriptapi.nim]: Done.

File: src/nimblepkg/nimscriptapi.nim (hunk 1 above)
Comment (kraptor) [reply src/nimblepkg/nimscriptapi.nim]: Done. I wanted to avoid an extra loop, but normally there are few tasks, so not critical.

**Comment by kraptor** [comment]
Ok, I see what is failing... let me fix printSeqIfLen template...

**Comment by kraptor** [comment]
I squashed the changes for easy review @dom96, everything seems ok now.

**Pull request merged**